pkg/alertmanager/state_replication.go

```diff
@@ -167,7 +174,26 @@ func (s *state) starting(ctx context.Context) error {
 		}
 	}
 
-	level.Info(s.logger).Log("msg", "state not settled but continuing anyway", "err", err)
+	level.Info(s.logger).Log("msg", "state not settled; trying to read from storage", "err", err)
+
+	// Attempt to read the state from persistent storage instead.
+	storeReadCtx, cancel := context.WithTimeout(ctx, s.storeReadTimeout)
+	defer cancel()
+
+	fullState, err := s.store.GetFullState(storeReadCtx, s.userID)
+	if err == alertspb.ErrNotFound {
```

Comment:
Always use `errors.Is()` instead of `==` to compare errors, because error could have been wrapped.

```suggestion
	if errors.Is(err, alertspb.ErrNotFound) {
```